src/TimeGridEvent.js

```diff
@@ -3,73 +3,93 @@ import cn from 'classnames'
-function TimeGridEvent(props) {
-  const {
-    style,
-    className,
-    event,
-    accessors,
-    isRtl,
-    selected,
-    label,
-    continuesEarlier,
-    continuesLater,
-    getters,
-    onClick,
-    onDoubleClick,
-    components: { event: Event, eventWrapper: EventWrapper },
-  } = props
-  let title = accessors.title(event)
-  let tooltip = accessors.tooltip(event)
-  let end = accessors.end(event)
-  let start = accessors.start(event)
-  let icon = event.icon
+class TimeGridEvent extends React.Component {
+  state = {}
 
-  let userProps = getters.eventProp(event, start, end, selected)
+  render() {
+    const {
+      style,
+      className,
+      event,
+      accessors,
+      isRtl,
+      selected,
+      label,
+      continuesEarlier,
+      continuesLater,
+      getters,
+      onClick,
+      onDoubleClick,
+      components: { event: Event, eventWrapper: EventWrapper },
+    } = this.props
+    let title = accessors.title(event)
+    let tooltip = accessors.tooltip(event)
+    let end = accessors.end(event)
+    let start = accessors.start(event)
+    let icon = event.icon
 
-  let { height, top, width, xOffset } = style
-  const inner = [
-    <div key="1" className="rbc-event-label">
-      {[
-        icon && (
-          <i className="material-icons" key={icon}>
-            {icon}
-          </i>
-        ),
-        event.label || label,
-      ]}
-    </div>,
-    <div key="2" className="rbc-event-content">
-      {Event ? <Event event={event} title={title} /> : title}
-    </div>,
-  ]
+    let userProps = getters.eventProp(event, start, end, selected)
 
-  return (
-    <EventWrapper type="time" {...props}>
-      <div
-        onClick={onClick}
-        onDoubleClick={onDoubleClick}
-        style={_.defaults({}, userProps.style, {
-          top: `${top}%`,
-          height: `${height}%`,
-          [isRtl ? 'right' : 'left']: `${Math.max(0, xOffset)}%`,
-          width: `${width}%`,
-        })}
-        title={
-          tooltip
-            ? (typeof label === 'string' ? label + ': ' : '') + tooltip
-            : undefined
-        }
-        className={cn('rbc-event', className, userProps.className, {
-          'rbc-selected': selected,
-          'rbc-event-continues-earlier': continuesEarlier,
-          'rbc-event-continues-later': continuesLater,
-        })}
-      >
-        {inner}
-      </div>
-    </EventWrapper>
-  )
+    let { height, top, width, xOffset } = style
+    const inner = [
+      <div key="1" className="rbc-event-label">
+        {[
+          icon && (
+            <i className="material-icons" key={icon}>
+              {icon}
+            </i>
+          ),
+          event.label || label,
+        ]}
+      </div>,
+      <div key="2" className="rbc-event-content">
+        {Event ? <Event event={event} title={title} /> : title}
+      </div>,
+    ]
+
+    // treat < 400 ms touches on events as clicks on touch devices
+    let { timestamp } = this.state
```

Comment:
fyi, this is actual fix: I've added two functions on start and end of touch events (`pointerDown` / `pointerUp`).
It fires click event when time between down and up is less than 400 ms - I checked empirically and to me the time was sufficient to determine click on touch device in that way.

Rest of changes in this file (except of attaching these functions to event handlers on 68-69 lines) is just changing the component to class